include/verilated.cpp

```diff
@@ -2506,22 +2506,56 @@ VerilatedContext::Serialized::Serialized() {
 
 bool VerilatedContext::assertOn() const VL_MT_SAFE {
     const VerilatedLockGuard lock{m_mutex};
-    return m_s.m_assertOn;
+    return m_s.m_assertOn.any();
 }
 void VerilatedContext::assertOn(bool flag) VL_MT_SAFE {
     const VerilatedLockGuard lock{m_mutex};
-    // set all assert types to 'on' when true, set all types to 'off' when false
-    m_s.m_assertOn = std::numeric_limits<VerilatedAssertType_t>::max()
-                     * static_cast<VerilatedAssertType_t>(flag);
-}
-bool VerilatedContext::assertOnGet(VerilatedAssertType_t flags) const VL_MT_SAFE {
-    return m_s.m_assertOn & flags;
+    if (flag) {
+        m_s.m_assertOn.set();
+    } else {
+        m_s.m_assertOn.reset();
+    }
 }
-void VerilatedContext::assertOnSet(VerilatedAssertType_t flags) VL_MT_SAFE {
-    m_s.m_assertOn |= flags;
+bool VerilatedContext::assertOnGet(VerilatedAssertType_t type,
+                                   VerilatedAssertDirectiveType_t directive) const {
+    const VerilatedLockGuard lock{m_mutex};
+    // Check if selected directive type bit in the assertOn is enabled for assertion type.
+    // Note: it is assumed that this is checked only for one type at the time.
+
+    // Flag unspecified assertion types as disabled.
+    if (type == 0) return false;
+
+    // Get index of 3-bit group guarding assertion type status.
+    const uint32_t typeMaskPosition = VL_CTZ_I(type);
+    // Check if directive type bit is enabled in corresponding assertion type bits.
+    const std::bitset<ASSERT_ON_WIDTH> assertsOn
+        = m_s.m_assertOn
+          & std::bitset<ASSERT_ON_WIDTH>(directive
+                                         << (typeMaskPosition * ASSERT_DIRECTIVE_TYPE_MASK_WIDTH));
```

Comment:
Some places seem to use bitset and other places using and/or of bit operations.  Would suggest just use a CData and use VL_BIT_ operations instead.

Reply:
I replaced `bitset` with `uint32_t`. Since, the whole algorithm is based on the assumption that only 25-bits are used in this variable, I think it will be more readable to use fixed width integer for it. Moreover, only subset of bits is manipulated thus, functions `VL_BIT_* ` does not apply here (they use `VL_IDATASIZE` when the used width is `ASSERT_ON_WIDTH`).

This integer width constrain greatly simplifies `assertOn()` function.